aiplatform/src/main/java/aiplatform/EmbeddingModelTuningSample.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package aiplatform;
+
+// [START aiplatform_sdk_embedding_model_tuning]
+import com.google.cloud.aiplatform.v1.CreatePipelineJobRequest;
+import com.google.cloud.aiplatform.v1.LocationName;
+import com.google.cloud.aiplatform.v1.PipelineJob;
+import com.google.cloud.aiplatform.v1.PipelineJob.RuntimeConfig;
+import com.google.cloud.aiplatform.v1.PipelineServiceClient;
+import com.google.cloud.aiplatform.v1.PipelineServiceSettings;
+import com.google.protobuf.Value;
+import java.io.IOException;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+// [END aiplatform_sdk_embedding_model_tuning]
+
+public class EmbeddingModelTuningSample {
+  public static void main(String[] args) throws IOException {
+    // [START aiplatform_sdk_embedding_model_tuning]
+    // TODO(developer): Replace these variables before running this sample.
+    String apiEndpoint = "us-central1-aiplatform.googleapis.com:443";
+    String project = "PROJECT";
+    String baseModelVersionId = "BASE_MODEL_VERSION_ID";
+    String taskType = "TASK_TYPE";
+    String pipelineJobDisplayName = "PIPELINE_JOB_DISPLAY_NAME";
+    String outputDir = "OUTPUT_DIR";
+    String queriesPath = "QUERIES";
+    String corpusPath = "CORPUS";
+    String trainLabelPath = "TRAIN_LABEL";
+    String testLabelPath = "TEST_LABEL";
+    int batchSize = 50;
+    int iterations = 300;
+
+    createEmbeddingModelTuningPipelineJob(
+        apiEndpoint,
+        project,
+        baseModelVersionId,
+        taskType,
+        pipelineJobDisplayName,
+        outputDir,
+        queriesPath,
+        corpusPath,
+        trainLabelPath,
+        testLabelPath,
+        batchSize,
+        iterations);
+    // [END aiplatform_sdk_embedding_model_tuning]
```

Comment:
Can you define a function with default values in Java? This pattern has been there for almost all the samples.

Reply:
There are no reasonable defaults for batch size and iterations. You cannot know what machine learning problems and downstream tasks your future customers will train a machine learning model for.

Reply:
Unfortunately I don't think Java supports default values. The only way I'm aware of would be through overloading, but that would make it even harder to read.

As for giving reasonable defaults, I think that can be done. I know they can change widely from case to case, but we can at least give a reasonable value for this specific example. What people are looking for is if the batch size should be something like 10, or 64, or 512, or 100,000. Even if it's not the most optimized, a reasonable default could be something that makes for reasonable enough performance while keeping memory usage low. Iterations can, again, be something that balances compute time with overall accuracy (however that's measured).

For the specific case of a code sample, we would usually incline towards using less resources and taking less time to complete so that users can play with the code without it being too expensive or resource intensive. For example, I usually try to keep my sample training jobs around 5 to 10 minutes, or at least less than 30 minutes. That would depend if, for example, after 5 minutes the results still look like noise, I would go higher, but always trying to keep it as fast as possible. I also like to give an example of what the results look like after training for longer periods like several hours so that they can see the results without having to go through all the hassle.

If it's something very important for this sample, there should be an explicit note both as comments in the code and in the README file.